tools/autograd/gen_variable_type.py

```diff
@@ -292,10 +296,57 @@
 #endif
 """)
 
+FW_DERIVATIVE_CHECK_TEMPLATE = CodeTemplate("""\
+isFwGradDefined(${req_inp})\
+""")
+
+FW_DERIVATIVE_DEFINED_GRAD_TEMPLATE = CodeTemplate("""\
+auto ${inp}_t_raw = toLegacyFwGrad(${inp});
```

Comment:
Why is it called Legacy from the getgo (do I repeat myself? Very well, I repeat myself)